src/core/toga/widgets/table.py

```diff
@@ -136,3 +136,79 @@ def on_select(self, handler):
         """
         self._on_select = wrapped_handler(self, handler)
         self._impl.set_on_select(self._on_select)
+
+    def add_column(self, heading, accessor=None):
+        """
+        Add a new column to the table
+
+        :param heading:     title of the column
+        :type heading:      ``string``
+        :param accessor:    accessor of this new column
+        :type heading:      ``string``
+        """
+
+        if not accessor:
+            accessor = to_accessor(heading)
+
+        if accessor in self._accessors:
+            raise ValueError('Column name "{}" already exits'.format(accessor))
```

Comment:
Minor nitpick on the error message - the problem is that the *accessor* is already in use, not the column name.